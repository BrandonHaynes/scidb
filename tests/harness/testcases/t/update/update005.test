--setup
--start-query-logging

--aql create  array expr2 <a:int64, b:int64>[x=1:5,2,0, y=1:5,2,0]
--shell --store-all --command "iquery -c $IQUERY_HOST -p $IQUERY_PORT -q "load expr2 from '${TEST_DATA_DIR}/expr2.txt'""

--test

--aql update expr2 set a=888 where x=5 and y=5

--aql update expr2 set a=777 where a=1

--aql update expr2 set a=555 where b=x+2

--aql update expr2 set a=1,b=2 where a+1=b-1

--aql update expr2 set b=6 where y=4

--aql update expr2 set b=0 where a=1 and b=6 and x=3


--cleanup
--aql drop array expr2

--stop-query-logging
