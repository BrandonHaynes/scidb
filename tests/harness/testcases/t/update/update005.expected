SCIDB QUERY : <create  array expr2 <a:int64, b:int64>[x=1:5,2,0, y=1:5,2,0]>
Query was executed successfully

SCIDB QUERY : <iquery -c $IQUERY_HOST -p $IQUERY_PORT -q "load expr2 from '${TEST_DATA_DIR}/expr2.txt'">
{x,y} a,b
{1,1} 1,3
{1,2} 5,7
{2,1} 21,23
{2,2} 25,27
{1,3} 9,11
{1,4} 13,15
{2,3} 29,31
{2,4} 33,35
{1,5} 17,19
{2,5} 37,39
{3,1} 41,43
{3,2} 45,47
{4,1} 61,63
{4,2} 65,67
{3,3} 49,51
{3,4} 53,55
{4,3} 69,71
{4,4} 73,75
{3,5} 57,59
{4,5} 77,79
{5,1} 81,83
{5,2} 85,87
{5,3} 89,91
{5,4} 93,95
{5,5} 97,99

SCIDB QUERY : <update expr2 set a=888 where x=5 and y=5>
{x,y} a,b
{1,1} 1,3
{1,2} 5,7
{2,1} 21,23
{2,2} 25,27
{1,3} 9,11
{1,4} 13,15
{2,3} 29,31
{2,4} 33,35
{1,5} 17,19
{2,5} 37,39
{3,1} 41,43
{3,2} 45,47
{4,1} 61,63
{4,2} 65,67
{3,3} 49,51
{3,4} 53,55
{4,3} 69,71
{4,4} 73,75
{3,5} 57,59
{4,5} 77,79
{5,1} 81,83
{5,2} 85,87
{5,3} 89,91
{5,4} 93,95
{5,5} 888,99

SCIDB QUERY : <update expr2 set a=777 where a=1>
{x,y} a,b
{1,1} 777,3
{1,2} 5,7
{2,1} 21,23
{2,2} 25,27
{1,3} 9,11
{1,4} 13,15
{2,3} 29,31
{2,4} 33,35
{1,5} 17,19
{2,5} 37,39
{3,1} 41,43
{3,2} 45,47
{4,1} 61,63
{4,2} 65,67
{3,3} 49,51
{3,4} 53,55
{4,3} 69,71
{4,4} 73,75
{3,5} 57,59
{4,5} 77,79
{5,1} 81,83
{5,2} 85,87
{5,3} 89,91
{5,4} 93,95
{5,5} 888,99

SCIDB QUERY : <update expr2 set a=555 where b=x+2>
{x,y} a,b
{1,1} 555,3
{1,2} 5,7
{2,1} 21,23
{2,2} 25,27
{1,3} 9,11
{1,4} 13,15
{2,3} 29,31
{2,4} 33,35
{1,5} 17,19
{2,5} 37,39
{3,1} 41,43
{3,2} 45,47
{4,1} 61,63
{4,2} 65,67
{3,3} 49,51
{3,4} 53,55
{4,3} 69,71
{4,4} 73,75
{3,5} 57,59
{4,5} 77,79
{5,1} 81,83
{5,2} 85,87
{5,3} 89,91
{5,4} 93,95
{5,5} 888,99

SCIDB QUERY : <update expr2 set a=1,b=2 where a+1=b-1>
{x,y} a,b
{1,1} 555,3
{1,2} 1,2
{2,1} 1,2
{2,2} 1,2
{1,3} 1,2
{1,4} 1,2
{2,3} 1,2
{2,4} 1,2
{1,5} 1,2
{2,5} 1,2
{3,1} 1,2
{3,2} 1,2
{4,1} 1,2
{4,2} 1,2
{3,3} 1,2
{3,4} 1,2
{4,3} 1,2
{4,4} 1,2
{3,5} 1,2
{4,5} 1,2
{5,1} 1,2
{5,2} 1,2
{5,3} 1,2
{5,4} 1,2
{5,5} 888,99

SCIDB QUERY : <update expr2 set b=6 where y=4>
{x,y} a,b
{1,1} 555,3
{1,2} 1,2
{2,1} 1,2
{2,2} 1,2
{1,3} 1,2
{1,4} 1,6
{2,3} 1,2
{2,4} 1,6
{1,5} 1,2
{2,5} 1,2
{3,1} 1,2
{3,2} 1,2
{4,1} 1,2
{4,2} 1,2
{3,3} 1,2
{3,4} 1,6
{4,3} 1,2
{4,4} 1,6
{3,5} 1,2
{4,5} 1,2
{5,1} 1,2
{5,2} 1,2
{5,3} 1,2
{5,4} 1,6
{5,5} 888,99

SCIDB QUERY : <update expr2 set b=0 where a=1 and b=6 and x=3>
{x,y} a,b
{1,1} 555,3
{1,2} 1,2
{2,1} 1,2
{2,2} 1,2
{1,3} 1,2
{1,4} 1,6
{2,3} 1,2
{2,4} 1,6
{1,5} 1,2
{2,5} 1,2
{3,1} 1,2
{3,2} 1,2
{4,1} 1,2
{4,2} 1,2
{3,3} 1,2
{3,4} 1,0
{4,3} 1,2
{4,4} 1,6
{3,5} 1,2
{4,5} 1,2
{5,1} 1,2
{5,2} 1,2
{5,3} 1,2
{5,4} 1,6
{5,5} 888,99

SCIDB QUERY : <drop array expr2>
Query was executed successfully

